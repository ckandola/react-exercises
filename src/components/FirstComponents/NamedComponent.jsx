import React, { Component } from "react";
import PropTypes from 'prop-types';

class NamedComponent extends Component {
  render() {
    return <p>My name is {this.props.name}</p>;
  }
}

export default NamedComponent;

NamedComponent.propTypes = {
  name: PropTypes.string,
};
