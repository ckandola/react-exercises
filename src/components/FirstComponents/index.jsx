import React from 'react';

import FirstComponent from './FirstComponent';
import SecondComponent from './SecondComponent';
import NamedComponent from './NamedComponent';

const FirstComponents = ({name = 'Anonymous'}) => {
    return (
        <div>
            <FirstComponent />    
            <SecondComponent />
            <NamedComponent name={name} />
        </div>
    );
};

export default FirstComponents;
