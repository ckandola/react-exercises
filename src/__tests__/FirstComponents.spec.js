import React from 'react';
import renderer from 'react-test-renderer';
import FirstComponents from '../components/FirstComponents';
import NamedComponent from '../components/FirstComponents/NamedComponent';

describe('First components test', () => {
    it('renders the default set of components correctly', () => {
        const firstComponents = renderer.create(<FirstComponents />);
        const tree = firstComponents.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('sets the NamedComponent name when it is given', () => {
        const namedComponent = renderer.create(<NamedComponent name="given" />);
        const instance = namedComponent.root;
        const name = instance.props.name;
        expect(name).toBe("given");
    });
});
