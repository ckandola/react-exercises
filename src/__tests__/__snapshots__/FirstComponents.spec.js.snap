// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First components test renders the default set of components correctly 1`] = `
<div>
  <h1>
    My very first component
  </h1>
  <h2>
    My second component
  </h2>
  <p>
    My name is 
    Anonymous
  </p>
</div>
`;
